dagshub/upload/errors.py

```diff
@@ -0,0 +1,87 @@
+import functools
+from dataclasses import dataclass
+
+import httpx
+
+
+@dataclass
+class UploadErrorResponseContent:
+    error: str
+    details: str
+
+
+@dataclass
+class GenericAPIErrorContent:
+    message: str
+
+
+_error_lookup = {}
+
+
+def _error_name(error: str):
```

Comment:
From [PEP8](https://peps.python.org/pep-0008/#descriptive-naming-styles):

>_single_leading_underscore: weak “internal use” indicator. E.g. from M import * does not import objects whose names start with an underscore.

This is basically internal to this file and I use this convention to indicate it as such

Reply:
What I meant is why make it internel. 
BTW I would a pattern like:
```python
@_register_exception_lookup(error_name="missing last_commit")
```
a lot more readable. This or simply writing the lookup table in plain text. 
WDYT?

Reply:
Not a fan of plain text because it's not really "expandable".
Basically I like the decorator pattern because if we need to add a new error, you throw on the decorator and don't need to remember where the lookup dict actually is, or where is it being used.

The naming correction is valid.

Reply:
Ok, how about we make it not internal and call it `register_upload_error(error_value="missing last_commit")`?

And `determine_error` will be `determine_upload_api_error` (since Python uses errors instead of exceptions, but also I want to make it obvious it's a function and not an error that is named wrong)